kyaml/yaml/merge3/merge3_test.go

```diff
@@ -15,24 +15,27 @@ var testCases = [][]testCase{scalarTestCases, listTestCases, mapTestCases, eleme
 
 func TestMerge(t *testing.T) {
 	for i := range testCases {
-		for _, tc := range testCases[i] {
-			actual, err := MergeStrings(tc.local, tc.origin, tc.update)
-			if tc.err == nil {
-				if !assert.NoError(t, err, tc.description) {
-					t.FailNow()
+		for j := range testCases[i] {
+			tc := testCases[i][j]
+			t.Run(tc.description, func(t *testing.T) {
+				actual, err := MergeStrings(tc.local, tc.origin, tc.update, !tc.noInfer)
```

Comment:
I see what was done here to change the behavior to opt-in, but this double negative may be confusing to future test authors.

Reply:
yeah, how about we fix it in a follow up?  it is basically a no-op change, but is going to generate a big diff